tools/frame_capture/src/image_copy.rs

```diff
@@ -0,0 +1,185 @@
+use std::sync::Arc;
+
+use bevy::prelude::*;
+use bevy::render::render_asset::RenderAssets;
+use bevy::render::render_graph::{self, NodeRunError, RenderGraph, RenderGraphContext};
+use bevy::render::renderer::{RenderContext, RenderDevice, RenderQueue};
+use bevy::render::{Extract, RenderApp};
+
+use bevy::render::render_resource::{
+    Buffer, BufferDescriptor, BufferUsages, CommandEncoderDescriptor, Extent3d, ImageCopyBuffer,
+    ImageDataLayout, MapMode,
+};
+use pollster::FutureExt;
+use wgpu::Maintain;
+
+use std::sync::atomic::{AtomicBool, Ordering};
+
+pub fn receive_images(
+    image_copiers: Query<&ImageCopier>,
+    mut images: ResMut<Assets<Image>>,
+    render_device: Res<RenderDevice>,
+) {
+    for image_copier in image_copiers.iter() {
+        if !image_copier.enabled() {
+            continue;
+        }
+        // Derived from: https://sotrh.github.io/learn-wgpu/showcase/windowless/#a-triangle-without-a-window
+        // We need to scope the mapping variables so that we can
+        // unmap the buffer
+        async {
+            let buffer_slice = image_copier.buffer.slice(..);
+
+            // NOTE: We have to create the mapping THEN device.poll() before await
+            // the future. Otherwise the application will freeze.
+            let (tx, rx) = futures_intrusive::channel::shared::oneshot_channel();
+            buffer_slice.map_async(MapMode::Read, move |result| {
+                tx.send(result).unwrap();
+            });
+            render_device.poll(Maintain::Wait);
+            rx.receive().await.unwrap().unwrap();
+            if let Some(image) = images.get_mut(&image_copier.dst_image) {
+                image.data = buffer_slice.get_mapped_range().to_vec();
+            }
+
+            image_copier.buffer.unmap();
+        }
+        .block_on();
+    }
+}
+
+pub const IMAGE_COPY: &str = "image_copy";
+
+pub struct ImageCopyPlugin;
+impl Plugin for ImageCopyPlugin {
+    fn build(&self, app: &mut App) {
+        let render_app = app
+            .add_systems(Update, receive_images)
+            .sub_app_mut(RenderApp);
+
+        render_app.add_systems(ExtractSchedule, image_copy_extract);
+
+        let mut graph = render_app.world.get_resource_mut::<RenderGraph>().unwrap();
+
+        graph.add_node(IMAGE_COPY, ImageCopyDriver::default());
+
+        graph.add_node_edge(IMAGE_COPY, bevy::render::main_graph::node::CAMERA_DRIVER);
```

Comment:
I think this line should be
```rust
graph.add_node_edge(bevy::render::main_graph::node::CAMERA_DRIVER, IMAGE_COPY);
```

or the buffer will be freed when still mapped and lead to an error